internal/store/horizontalpodautoscaler.go

```diff
@@ -192,22 +248,65 @@ func createHPASpecTargetMetric() generator.FamilyGenerator {
 			for _, m := range a.Spec.Metrics {
 				var metricName string
 				var metricTarget autoscaling.MetricTarget
+				var fullTargetName string
 				// The variable maps the type of metric to the corresponding value
 				metricMap := make(map[metricTargetType]float64)
 
 				switch m.Type {
 				case autoscaling.ObjectMetricSourceType:
 					metricName = m.Object.Metric.Name
 					metricTarget = m.Object.Target
+					fullTargetName = m.Object.DescribedObject.Name
+				default:
+					// Skip unsupported metric type
+					continue
+				}
+
+				if metricTarget.Value != nil {
+					metricMap[value] = convertValueToFloat64(metricTarget.Value)
+				}
+				if metricTarget.AverageValue != nil {
+					metricMap[average] = convertValueToFloat64(metricTarget.AverageValue)
+				}
+				if metricTarget.AverageUtilization != nil {
+					metricMap[utilization] = float64(*metricTarget.AverageUtilization)
+				}
+
+				for metricTypeIndex, metricValue := range metricMap {
+					ms = append(ms, &metric.Metric{
+						LabelKeys:   objectMetricLabels,
+						LabelValues: []string{metricName, metricTypeIndex.String(), fullTargetName},
+						Value:       metricValue,
+					})
+				}
+			}
+			return &metric.Family{Metrics: ms}
+		}),
+	)
+}
+
+func createHPASpecTargetMetric() generator.FamilyGenerator {
```

Comment:
nit: Unify createHPASpecTargetMetric and createHPASpecTargetObjectMetric. These two functions have  some similar codes.

For example 
- createHPASpecTargetMetric(CollectContainerResourceMetricSourceType=false)
- createHPASpecTargetMetric(CollectContainerResourceMetricSourceType=true)